src/client/QXmppVCardManager.cpp

```diff
@@ -113,7 +113,8 @@ bool QXmppVCardManager::handleStanza(const QDomElement& element)
         QXmppVCardIq vCardIq;
         vCardIq.parse(element);
 
-        if (vCardIq.from().isEmpty()) {
+        QString from = vCardIq.from();
+        if (from.isEmpty() || from == client()->configuration().jidBare()) {
```

Comment:
```suggestion
        if (vCardIq.from().isEmpty() || vCardIq.from() == client()->configuration().jidBare()) {
```

This way you don't need to copy the QString.